### PR TITLE
Google tag manager

### PR DESCRIPTION
**Pull request by Shiva656**

### CODE REVIEW

File: template.tpl [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+  {
+    "type": "TEXT",
+    "name": "api_key",
+    "displayName": "Enter the api key.",
+    "simpleValueType": true
+  },
+  {
+    "type": "TEXT",
+    "name": "region",
+    "displayName": "Enter the region.",
```
[the remark this answers is not in the document]
Comment (Shiva656) [reply template.tpl]: sir, encoded region in api key

File: metadata.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+homepage: "https://pushninja.com"
+documentation: ""
+versions:
+  - sha: d4ea375e97e48d98a11c37c13b76cdc652e3b347
```
[the remark this answers is not in the document]
Comment (Shiva656) [reply metadata.yaml:8]: yes sir after commiting the template when i have added in metadata file

File: template.tpl [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+  "type": "TAG",
+  "categories": "MARKETING",
+  "id": "cvt_temp_public_id",
+  "version": 1,
+  "securityGroups": [],
+  "displayName": "PushNinja",
+  "brand": {
+    "id": "brand_dummy",
+    "displayName": "",
+    "thumbnail": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAMAAABEpIrGAAABCFBMVEUAAAAAAP9VVf8rgNUkbdsggN85ceMzgOYudOgrgOopeuAngOIugOQrd+Ype+YufOgreOMpfOMufOUteeYtfeMteecrfOQteucseuQsfOQrfOUte+Ute+QseuQse+UreuUre+Ure+QteuQtfOUreuYte+YtfOQse+Qre+UrfOUte+QteuUte+Yre+Ute+YsfOQre+Ure+YreuYre+Qte+Ute+YsfOYsfOQse+Yse+Use+Use+Yre+Qse+Use+Use+UseuQse+Use+Use+Use+Use+Use+Use+UteuUse+Yse+Use+Use+Use+Use+Use+Use+Use+Use+Use+Use+Yse+Use+X///8y/OYWAAAAVnRSTlMAAQMGBwgJCgsYGRocHh8hJCUnKC0/QklLTE5PVVZXWFlfYGFkZmdoamtyd3iBg4aHjY6ZmqCho6qus7W2ubrHysvM2Nvg4uTl5uvs7fLz9Pb3+Pn6/s7+4qMAAAABYktHRFd9CtkfAAABAklEQVQYGb3BaTcCUQAG4HfKVLaUabEkxFiyLyHCjMJUI9W47///KTqnjHunxjl98TyYTiSTz2ciCJO8bnOgfbWIiQ77ZKNWa5C9fUxwSnGzjAHjVvAEYzbpFTGy9SUKCIg73IPvgO8xqEw+a/BpFneheuA2JDu8h+qNSUiW+AqF5okoJFHhaZAl2IGiwwRkc3SgaHIWMoN1KOo0IFvlIxRPXIHsmBdQXPIIsipLUJRYhWTmQyxAMS8+dfzaoI0Am0X4tBeaCDBpa/hhsqkjQHdoYiTbZaUwpsJuFkMthmhhqHwXooz/l3LPgHM3hTDpPi2LvTRCrbuku4Y/xHO5GKbzDRUBNgZtRihdAAAAAElFTkSuQmCC"
```
[the remark this answers is not in the document]
Comment (Shiva656) [reply template.tpl]: sir, its  an pushninja logo

File: template.tpl [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+const log = require('logToConsole');
+const injectScript = require('injectScript');
+const setInWindow = require('setInWindow');
+const encodeUriComponent = require('encodeUriComponent');
+const queryPermission = require('queryPermission');
+
+// Reading the value's from fields
+const api_key = encodeUriComponent(data.api_key);
+const region = encodeUriComponent(data.region);
+// Setting apikey to global variable.
```
[the remark this answers is not in the document]
Comment (Shiva656) [reply template.tpl]: removed the region and global variables.

File: template.tpl [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+if (queryPermission('access_globals', 'readwrite', '_push')) {
+  setInWindow("_push",{"apiKey": api_key}, true);
+}
+// Setting region to global variable.
+if (queryPermission('access_globals', 'readwrite', '_pushfcm')) {
+  setInWindow("_pushfcm", {'location': region}, true);
+}
+log('data = verify', data);
+const trackingUrl='https://infinity-public-js.500apps.com/push/push.min.js';
+injectScript(trackingUrl,data.gtmOnSuccess,data.gtmOnFailure);
```
[the remark this answers is not in the document]
Comment (Shiva656) [reply template.tpl:54]: sir, added the wrapper as like hubspot

File: template.tpl [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+Template Gallery Developer Terms of Service available at
+https://developers.google.com/tag-manager/gallery-tos (or such other URL as
+Google may provide), as modified from time to time.
+
+
+___INFO___
+
+{
+  "type": "TAG",
+  "categories": "MARKETING",
```
[the remark this answers is not in the document]
Comment (Shiva656) [reply template.tpl:13]: yes sir, in clever push & wonder push they are using same category

**Comment by agilecrm** [comment]
Good job. This was long pending and we have achieved it.
Can we see a video end to end before it gets merged and we do the same for other apps?

**Comment by Shiva656** [comment]
> Good job. This was long pending and we have achieved it.
> Can we see a video end to end before it gets merged and we do the same for other apps?

Sir, here is the end to end video https://drive.google.com/file/d/1IXjRJ45iM0okn57-NooTYcftacU5X-ex/view

**Comment by agilecrm** [comment]
GTM code changes in the script:

Google Tag Manager

If you are using Google tag manager, you can search for "PushNinja" and use the following key while prompted.
{{xxx}}

-----
The code is a label and not field. It should be in bold also.

**Comment by agilecrm** [comment]
Google tag changes:
Enter the api key. ->PushNinja Key:


image

? help text is possible as seen in the image:

Please copy the API key when you have added a script in the 500apps portal

**Comment by agilecrm** [comment]
With these changes, we are good to go

**Comment by Shiva656** [comment]
Sir, fixed the above comments here is the video https://drive.google.com/file/d/1Gdp93Wv-M2civJTTpznxSaV4ckiExWo_/view

**Comment by agilecrm** [comment]
Good job Siva

**Pull request merged**